dev/com.ibm.ws.jca/src/com/ibm/ws/jca/internal/ActivationConfig.java

```diff
@@ -41,6 +65,88 @@ public ActivationConfig(Properties activationConfigProps, String destinationRef,
         this.applicationName = appName;
     }
 
+    /*
+     * This method rereates the CMConfigData Object from a stream - all the members will be
+     * re-initialized.
+     */
+    @SuppressWarnings("unchecked")
+    private void readObject(ObjectInputStream stream) throws IOException, ClassNotFoundException {
+
+        /*
+         * Since this is a simple class all we have to do is read each member from the stream.
+         * the order has to remain identical to writeObject, so I've just gone alphabetically.
+         *
+         * If we change the serialversionUID, we may need to modify this method as well.
+         */
+
+        if (tc.isEntryEnabled()) {
+            Tr.entry(this, tc, "readObject", stream);
+        }
+
+        ObjectInputStream.GetField getField = stream.readFields();
+
+        if (tc.isDebugEnabled()) {
+
+            for (int i = 0; i < serialPersistentFields.length; i++) {
+
+                String fieldName = serialPersistentFields[i].getName();
+
+                if (getField.defaulted(fieldName))
+                    Tr.debug(this, tc, "DESERIALIZATION_FIELD_NOT_FOUND_J2CA0278", fieldName, getClass().getName());
+
+            }
+
+        }
+
+        activationConfigProps = (Properties) getField.get("activationConfigProps", null);
+        authenticationAlias = (String) getField.get("authenticationAlias", null);
+        qmid = (String) getField.get("qmid", null);
+
+        if (tc.isEntryEnabled())
+            Tr.exit(this, tc, "readObject activation config - " + this.toString());
+    }
+
+    /*
+     * this method writes the CMConfigDataOjbect to a stream
+     */
+    private void writeObject(ObjectOutputStream stream) throws IOException {
+
+        /*
+         * Since this is a simple class all we have to do is write each member to the stream.
+         * the order has to remain identical to readObject, so I've just gone alphabetically.
+         *
+         * If we change the serialversionUID, we may need to modify this method as well.
+         */
+
+        if (tc.isEntryEnabled()) {
+            Tr.entry(this, tc, "writeObject", stream);
+        }
+
+        ObjectOutputStream.PutField putField = stream.putFields();
+
+        putField.put("activationConfigProps", activationConfigProps);
+        putField.put("authenticationAlias", authenticationAlias);
+        putField.put("qmid", qmid);
+
+        stream.writeFields();
+
+        if (tc.isEntryEnabled()) {
+            Tr.exit(this, tc, "writeObject activation config - " + this.toString());
+        }
+
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see java.lang.Object#toString()
+     */
+    @Override
+    public String toString() {
+        return "ActivationConfig [activationConfigProps=" + activationConfigProps + ", destinationRef=" + destinationRef + ", authenticationAlias=" + authenticationAlias
```

Comment:
might not want to include the actionConfigProps value in case it is very lengthy or has passwords in it (is that possible?)